hypervisor/src/lib.rs

```diff
@@ -25,6 +25,8 @@ extern crate thiserror;
 #[macro_use]
 extern crate anyhow;
 
+pub use kvm::KvmHypervisor as Hypervisor;
```

Comment:
I don't see the reason why we would do that. It's important that we keep Hypervisor as the trait, and then HypervHypervisor and KvmHypervisor will implement it. The caller (vmm crate) will initialize the right structure depending on the rust feature present at build time. Once initialized, the structure will be used as the Hypervisor trait for the generic code shared by both features (kvm and hyperv).

Reply:
But that means code in VMM crate will need to have 

```
#[cfg(feature = "kvm")]
let hv = hypervisor::kvm::KvmHypervisor::new().unwrap();
#[cfg(feature = "hyperv")]
let hv = hypervisor::hyperv::HyperVHypervisor::new().unwrap();
let hv: Arc<dyn hypervisor::Hypervisor> = Arc::new(hv);
```

And this snippet needs to be duplicated in _all_ the present and future test cases. That looks rather cumbersome, isn't it?

Reply:
No that's fine. From the vmm crate we want to be able to identify what's kvm vs hyperv specific. We don't want this kind of thing to be pushed down to the hypervisor crate.

Reply:
Sure. This is also fine by me.